dms/models/dms_directory.py

```diff
@@ -0,0 +1,806 @@
+        return res
+
+    @api.depends("res_model")
+    def _compute_model_id(self):
+        for record in self:
+            if not record.res_model:
+                record.model_id = False
+                continue
+            record.model_id = (
+                self.env["ir.model"].sudo().search([("model", "=", record.res_model)])
+            )
+
+    def _inverse_model_id(self):
+        for record in self:
+            record.res_model = record.model_id.model
+
+    @api.depends("name", "parent_id.complete_name")
+    def _compute_complete_name(self):
+        for category in self:
+            if category.parent_id:
+                category.complete_name = "{} / {}".format(
+                    category.parent_id.complete_name,
+                    category.name,
+                )
+            else:
+                category.complete_name = category.name
+
+    @api.depends("parent_id")
+    def _compute_storage_id(self):
+        for record in self:
+            if record.parent_id:
+                record.storage_id = record.parent_id.storage_id
+            else:
+                # HACK: Not needed in v14 due to odoo/odoo#64359
+                record.storage_id = record.storage_id
+
+    @api.depends("user_star_ids")
+    def _compute_starred(self):
+        for record in self:
+            record.starred = self.env.user in record.user_star_ids
+
+    @api.depends("child_directory_ids")
+    def _compute_count_directories(self):
+        for record in self:
+            directories = len(record.child_directory_ids)
+            record.count_directories = directories
+            record.count_directories_title = _("%s Subdirectories") % directories
+
+    @api.depends("file_ids")
+    def _compute_count_files(self):
+        for record in self:
+            files = len(record.file_ids)
+            record.count_files = files
+            record.count_files_title = _("%s Files") % files
+
+    @api.depends("child_directory_ids", "file_ids")
+    def _compute_count_elements(self):
+        for record in self:
+            record.count_elements = record.count_files + record.count_directories
+
+    def _compute_count_total_directories(self):
+        for record in self:
+            count = (
+                self.search_count([("id", "child_of", record.id)]) if record.id else 0
+            )
+            record.count_total_directories = count - 1 if count > 0 else 0
+
+    def _compute_count_total_files(self):
+        model = self.env["dms.file"]
+        for record in self:
+            # Prevent error in some NewId cases
+            record.count_total_files = (
+                model.search_count([("directory_id", "child_of", record.id)])
+                if record.id
+                else 0
+            )
+
+    def _compute_count_total_elements(self):
+        for record in self:
+            record.count_total_elements = (
+                record.count_elements + record.count_total_directories
```

Comment:
Ok I understand. It was because portal user doesn't have read access to the file, therefore can't access the icon field.
I pushed a fix, you can try.